### PR TITLE
With this change, bucketing/padding of input is applied to health check.

### PR DESCRIPTION
**Pull request by srajabos**
# What does this PR do?

This PR applies bucketing/padding of input for proto messages in Causal LM 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@yuanwu2017 @mandy-li @tthakkal @skavulya

### CODE REVIEW

File: server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -460,11 +460,11 @@ def from_pb(
         tokenizer: PreTrainedTokenizerBase,
         dtype: torch.dtype,
         device: torch.device,
+        max_input_length: int = PAD_SEQUENCE_TO_MULTIPLE_OF,
```
Comment (yuanwu2017) [inline server/text_generation_server/models/causal_lm.py]: Remove the line.

File: server/text_generation_server/models/causal_lm.py [hunk 2]
```diff
@@ -460,11 +460,11 @@ def from_pb(
...
     ) -> "CausalLMBatch":
         dbg_trace('FROM_PB', f'num_reqs:{len(pb.requests)}')
         requests = [CausalLMRequest.from_pb(idx, req, tokenizer) for idx, req in enumerate(pb.requests)]
 
-        max_input_length = max(r.data.truncate for r in requests)
```
Comment (yuanwu2017) [inline server/text_generation_server/models/causal_lm.py:467]:
```
max_input_length = max(r.data.truncate for r in requests)
if max_input_length < PAD_SEQUENCE_TO_MULTIPLE_OF:
     max_input_length = PAD_SEQUENCE_TO_MULTIPLE_OF
```
Remove [line 505](https://github.com/huggingface/tgi-gaudi/blob/6ba3d1d6e58cd44e44f23d5e46d601648fea9442/server/text_generation_server/models/causal_lm.py#L505)

**Review by yuanwu2017 — CHANGES_REQUESTED** [review]
The inference request message doesn't include the max_input_length.

**Comment by srajabos** [comment]
fixed with minimal changes to existing code.

**Review by yuanwu2017 — APPROVED** [review]

**Comment by skavulya** [comment]
Thanks @srajabos  LGTM.

**Review by skavulya — APPROVED** [review]

**Comment by srajabos** [comment]
@mandy-li @regisss , please review and merge if ok for you.

**Comment by mandy-li** [comment]
@srajabos , pls address yuan's comments. if you already did, resolve the conversation

**Comment by srajabos** [comment]
Sure @mandy-li. Resolved the conversation.

**Review by tthakkal — APPROVED** [review]

**Review by schoi-habana — APPROVED** [review]
lgtm

**Comment by srajabos** [comment]
@mandy-li, @tthakkal , @schoi-habana can you merge the changes. Thanks.

**Review by regisss — APPROVED** [review]

**Pull request merged**